File: java/iterator/ej2/AllCandidates.java
package patrones.iterator.ej2;

import java.util.Enumeration;
import java.util.Iterator;
import java.util.Vector;

public class AllCandidates implements Iterator {
	private Vector data;
	Enumeration ec;
	Candidate nextCandidate;

	public AllCandidates() {
		initialize();
		ec = data.elements();
	}

	private void initialize() {
		/*
		 * Get data from db.
		 */
		FileUtil util = new FileUtil();
		data = util.fileToVector("Candidates.txt");

	}

	public boolean hasNext() {
		boolean matchFound = false;
		nextCandidate = null;
		while (ec.hasMoreElements()) {
			Candidate tempObj = (Candidate) ec.nextElement();
			nextCandidate = tempObj;
			break;
		}
		return (nextCandidate != null);
	}

	public Object next(){
		if (nextCandidate != null) {
			return nextCandidate;
		}
		return null;
	}

	public void remove() {
	}
}

File: c#/iterator/Ejm2/AllCandidates.cs
using System;
using System.Collections;
namespace Iterator.Ejm2
{
    /// <summary>
    /// Es un ConcreteIterator
    /// </summary>
    public class AllCandidates : Iterator
    {
        private ArrayList Data;
        IEnumerator Enum;
        Candidate nextCandidate;

        public AllCandidates()
        {
            Initialize();
            Enum = Data.GetEnumerator();
        }

        private void Initialize()
        {
            /*
             * Get data from db.
             */
            FileUtil util = new FileUtil();
            Data = util.FileToArrayList("Candidates.txt");

        }

        public override bool HasNext()
        {
            nextCandidate = null;
            while (Enum.MoveNext())
            {
                Candidate tempObj = (Candidate)Enum.Current;
                nextCandidate = tempObj;
                break;
            }
            return (nextCandidate != null);
        }

        public override Object Next()
        {
            if (nextCandidate != null)
            {
                return nextCandidate;
            }
            return null;
        }
    }
}
